lib/graphql/language/generation.rb

```diff
@@ -14,189 +14,10 @@ module Generation
-          out << " = " + node.types.map(&:name).join(" | ")
-        when Nodes::EnumTypeDefinition
-          out = generate_description(node)
-          out << "enum #{node.name}#{generate_directives(node.directives)} {\n"
-          node.values.each.with_index do |value, i|
-            out << generate_description(value, indent: '  ', first_in_block: i == 0)
-            out << generate(value)
-          end
-          out << "}"
-        when Nodes::EnumValueDefinition
-          out = "  #{node.name}".dup
-          out << generate_directives(node.directives)
-          out << "\n"
-        when Nodes::InputObjectTypeDefinition
-          out = generate_description(node)
-          out << "input #{node.name}"
-          out << generate_directives(node.directives)
-          out << " {\n"
-          node.fields.each.with_index do |field, i|
-            out << generate_description(field, indent: '  ', first_in_block: i == 0)
-            out << "  #{generate(field)}\n"
-          end
-          out << "}"
-        when Nodes::DirectiveDefinition
-          out = generate_description(node)
-          out << "directive @#{node.name}"
-          out << "(#{node.arguments.map { |a| generate(a) }.join(", ")})" if node.arguments.any?
-          out << " on #{node.locations.join(' | ')}"
-        when Nodes::AbstractNode
-          node.to_query_string(indent: indent)
-        when FalseClass, Float, Integer, NilClass, String, TrueClass
-          GraphQL::Language.serialize(node)
-        when Array
-          "[#{node.map { |v| generate(v) }.join(", ")}]".dup
-        when Hash
-          "{#{node.map { |k, v| "#{k}: #{generate(v)}" }.join(", ")}}".dup
-        else
-          raise TypeError
-        end
-      end
-
-      private
-
-      def generate_directives(directives)
-        if directives.any?
-          directives.map { |d| " #{generate(d)}" }.join
-        else
-          ""
-        end
-      end
-
-      def generate_selections(selections, indent: "")
-        if selections.any?
-          out = " {\n".dup
-          selections.each do |selection|
-            out << generate(selection, indent: indent + "  ") << "\n"
-          end
-          out << "#{indent}}"
-        else
-          ""
-        end
-      end
-
-      def generate_description(node, indent: '', first_in_block: true)
-        return ''.dup unless node.description
-
-        description = indent != '' && !first_in_block ? "\n".dup : "".dup
-        description << GraphQL::Language::Comments.commentize(node.description, indent: indent)
-      end
-
-      def generate_field_definitions(fields)
-        out = " {\n".dup
-        fields.each.with_index do |field, i|
-          out << generate_description(field, indent: '  ', first_in_block: i == 0)
-          out << "  #{generate(field)}\n"
-        end
-        out << "}"
+      def generate(node, indent: "", printer: GraphQL::Language::Printer)
+        printer.new(node).print(indent: indent)
```

Comment:
Hmm, maybe this code isn't tested? 

I see that `Printer` is initialized with `node`, but `#initialize` was removed in https://github.com/rmosolgo/graphql-ruby/pull/1118/commits/0b9ade35123286354faa351d34abdb3ec420d7fe#diff-c8deb24bd9b63b4eec0de9a19914a038L26. Can this API also accept a printer _instance_ now? I think you were 👌  on that suggestion.